metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_shares.py

```diff
@@ -0,0 +1,169 @@
+                logger.debug(f"database {db.name} is shared as outbound share.")
+
+            for schema in db.schemas:
+                for table_name in schema.tables + schema.views:
+                    # TODO: If this is outbound database,
+                    # 1. attempt listing shares using `show shares` to identify name of share associated with this database (cache query result).
+                    # 2. if corresponding share is listed, then run `show grants to share <share_name>` to identify exact tables, views included in share.
+                    # 3. emit siblings only for the objects listed above.
+                    # This will work only if the configured role has accountadmin role access OR is owner of share.
+                    # Otherwise ghost nodes may be shown in "Composed Of" section for tables/views in original database which are not granted to share.
+                    yield from self.get_siblings(
+                        db.name,
+                        schema.name,
+                        table_name,
+                        True if outbounds else False,
+                        sibling_dbs,
+                    )
+
+                    if inbound:
+                        # SnowflakeLineageExtractor is unaware of database->schema->table hierarchy
+                        # hence this lineage code is not written in SnowflakeLineageExtractor
+                        # also this is not governed by configs include_table_lineage and include_view_lineage
+                        yield self.get_upstream_lineage_with_primary_sibling(
+                            db.name, schema.name, table_name, inbound
+                        )
+
+        self.report_missing_databases(databases)
+
+    def get_sharing_details(self, db):
+        inbound = (
+            self.config.inbound_shares_map.get(db.name)
+            if self.config.inbound_shares_map
+            and db.name in self.config.inbound_shares_map
+            else None
+        )
+        outbounds = (
+            self.config.outbound_shares_map.get(db.name)
+            if self.config.outbound_shares_map
+            and db.name in self.config.outbound_shares_map
+            else None
+        )
+
+        return inbound, outbounds
+
+    def report_missing_databases(self, databases):
+        db_names = [db.name for db in databases]
+        missing_dbs = []
+        if self.config.inbound_shares_map:
+            missing_dbs.extend(
+                [db for db in self.config.inbound_shares_map if db not in db_names]
+            )
+        if self.config.outbound_shares_map:
+            missing_dbs.extend(
+                [db for db in self.config.outbound_shares_map if db not in db_names]
+            )
+        if missing_dbs:
+            self.report_warning(
+                "snowflake-shares",
+                f"Databases {missing_dbs} were not ingested. Siblings/Lineage will not be set for these.",
+            )
+
+    def get_siblings(
```

Comment:
Since this yields workunits, I'd vote for `gen_siblings` over `get_siblings`, same thing below, but don't feel too strongly